src/fs/metadata_log.cc

```diff
@@ -134,40 +166,54 @@ void metadata_log::memory_only_delete_inode(inode_t inode) {
     assert(not it->second.is_linked());
 
     std::visit(overloaded {
-        [](const inode_info::directory& dir) {
+        [](inode_info::directory& dir) {
             assert(dir.entries.empty());
         },
-        [](const inode_info::file&) {
-            // TODO: for compaction: update used inode_data_vec
+        [&](inode_info::file& file) {
+            cut_out_data_range(file, {
+                0,
+                std::numeric_limits<decltype(file_range::end)>::max()
+            });
+            ondisk_entry_size<ondisk_truncate>();
         }
     }, it->second.contents);
 
     _inodes.erase(it);
+    _ondisk_log_size += ondisk_entry_size<ondisk_delete_inode>();
+    _rewrite_log_size -= ondisk_entry_size<ondisk_create_inode>();
 }
 
 void metadata_log::memory_only_small_write(inode_t inode, file_offset_t file_offset, temporary_buffer<uint8_t> data) {
     inode_data_vec data_vec = {
         {file_offset, file_offset + data.size()},
         inode_data_vec::in_mem_data {std::move(data)}
     };
-
+    size_t log_increase = ondisk_entry_size<ondisk_small_write_header>(data.size());
     auto it = _inodes.find(inode);
     assert(it != _inodes.end());
     assert(it->second.is_file());
     write_update(it->second.get_file(), std::move(data_vec));
+    _ondisk_log_size += log_increase;
+    _rewrite_log_size += log_increase;
 }
 
 void metadata_log::memory_only_disk_write(inode_t inode, file_offset_t file_offset, disk_offset_t disk_offset,
         size_t write_len) {
+    assert(write_len <= _cluster_size);
+    size_t log_increase = write_len == _cluster_size ?
+        ondisk_entry_size<ondisk_large_write>() :
+        ondisk_entry_size<ondisk_medium_write>();
```

Comment:
```suggestion
    size_t log_increase = (write_len == _cluster_size ?
        ondisk_entry_size<ondisk_large_write>() :
        ondisk_entry_size<ondisk_medium_write>());
```
I think that `=` and `==` that are near one another may look deceptive. And it is error prone, consider:
```
size_t log_increase = write_len = _cluster_size ?
```
I would work (without warnings?).